### ISSUE TITLE
Issue while deploying report in app.powerbi.com

### ISSUE THREAD

**Issue by shaktimistry** [maikvandergaag/msft-extensions#114]
Hi,

We are trying to use this extension for the first time. But we keep getting below error.
"Cannot find an overload for "UserCredential" and the argument count: "2"."

We have app registered in Azure using App Registration. App has access to dev ops and Power BI. Also we are using Power BI Pro account.

![Publish_PBIX](https://user-images.githubusercontent.com/56104050/66100226-b76ae780-e5c7-11e9-9f57-2f927b8c54d0.png)

Can you please help us understand what we are doing wrong?

**Comment by maikvandergaag** [comment]
Hi,

Microsoft made some changes to the authentication libraries. Are you using a hosted agent or your own build agent.

**Comment by shaktimistry** [comment]
We are using own build agent.

**Comment by maikvandergaag** [comment]
If you are using your own build agent. It may be a problem with the version of the authentication library you have running on the server. At this moment I am working on a fix that should resolve it.

**Comment by maikvandergaag** [comment]
if needed we have a debug version ready that I can share with you. This version also support service principal authentication

**Comment by shaktimistry** [comment]
That would be great. We can work with debug version for now. 

Thank you so much for the quick response.

**Comment by maikvandergaag** [comment]
Then I will need your organization id

**Comment by shaktimistry** [comment]
Apologies. We were having credential related issue when we were trying to use Publish Power BI action with Task version 1. Also, I will have to check regarding organization id.

But with Power BI Actions task version 3.0 we are going getting work space setting. But after that we are getting below error.  Can you please help us with resolution?

2019-10-04T04:18:54.2234289Z Getting AAD Token for user: ***
2019-10-04T04:18:54.2234289Z Publishing PowerBI FIle: ********, in workspace: Development with user: ***
2019-10-04T04:18:54.2234289Z Getting Power BI Workspace properties; Development
2019-10-04T04:18:54.3794299Z ##[error]The 'Content-Type' header must be modified using the appropriate property or method.
Parameter name: name

**Comment by maikvandergaag** [comment]
Can you show the configured parameters and are you sure the account that you use has the appropriate permissions

**Comment by shaktimistry** [comment]
Hey I was able to resolve the issue and upload report to app.powerbi.com using azure.

I removed the content-type from authorization header and passed to Invoke-RestMethod as param.

  $apiHeaders = @{       
        'Authorization'= "Bearer $AccessToken"
    }

$result = Invoke-RestMethod -Uri $Url -Headers $apiHeadersGet -Method $Method -ContentType "application/json"

Thank you so much for making such a wonderful extension. It would be a big help to upload Power BI report without opening the same.
